semanticdb/semanticdb3/guide.md

```diff
@@ -0,0 +1,576 @@
+      <br/><br/>
+      NOTE: Some of our users <a href="https://github.com/scalameta/scalameta/issues/1398">have reported deadlocks</a> supposedly caused
+      by enabling <code>--par</code>. Proceed at your own risk.
+    </td>
+    <td>
+      <code>--no-par</code>
+    </td>
+  </tr>
+</table>
+
+## Consuming SemanticDB
+
+### Scalameta
+
+Scalameta includes the `semanticdb3` library that contains
+[ScalaPB](https://scalapb.github.io/) bindings to
+[the SemanticDB protobuf schema](semanticdb3.proto). Using this library,
+one can model SemanticDB entities as Scala case classes and serialize/deserialize
+them into bytes and streams. For more information, check out [the Scaladoc](https://static.javadoc.io/org.scalameta/semanticdb3_2.11/3.7.4/index.html#scala.meta.internal.semanticdb3.TextDocuments).
+
+NOTE: At this point, it's not very easy to take a classpath, traverse it and
+load all available SemanticDB payloads. SemanticDB-based tools are currently
+responsible for implementing discovery and deserialization on their own.
+However, we recognize this as a usability problem and are planning to
+[improve the situation](https://github.com/scalameta/scalameta/issues/1566)
+very soon.
+
+For examples of developer tools that consume SemanticDB, take a look at:
+  * [Scalafix](https://github.com/scalacenter/scalafix/)
+  * [Metadoc](https://github.com/scalameta/metadoc)
+  * [Metals](https://github.com/scalameta/metals)
+
+### Protoc
+
+The Protocol Compiler tool (`protoc`) can inspect protobuf payloads in
+`--decode` (takes a schema) and `--decode_raw` (doesn't need a schema) modes.
+For the reference, here's [the SemanticDB protobuf schema](semanticdb3.proto).
+
+```
+$ tree
+.
+├── META-INF
+│   └── semanticdb
+│       └── Test.scala.semanticdb
+└── Test.scala
+
+$ protoc --proto_path <directory with the .proto file>\
+--decode scala.meta.internal.semanticdb3.TextDocuments\
+semanticdb3.proto < META-INF/semanticdb/Test.scala.semanticdb
+
+documents {
+  schema: SEMANTICDB3
+  uri: "Test.scala"
+  text: "object Test {\n  def main(args: Array[String]): Unit = {\n    println(\"hello world\")\n  }\n}\n"
+  symbols {
+    symbol: "_empty_.Test."
+    kind: OBJECT
+    properties: 8
+    name: "Test"
+    tpe {
+      tag: TYPE_REF
+      typeRef {
+        symbol: "_empty_.Test."
+      }
+    }
+    accessibility {
+      tag: PUBLIC
+    }
+    owner: "_empty_."
+    language: SCALA
+  }
+  ...
+}
+```
+
+`protoc` was useful for getting things done in the early days of SemanticDB,
+but nowadays it's a bit too low-level. In order to provide SemanticDB users
+more control over prettyprinting and automate classpath traversal, we developed
+Metap.
+
+### Metap
+
+Metap is a command-line tool that takes a list of paths and then prettyprints
+all .semanticdb files that it finds in these paths. Advanced options control
+prettyprinting format.
+
+```bash
+alias metap="coursier launch org.scalameta:metap_2.11:3.7.4 --"
```

Comment:
Should we include native installation instructions like here https://github.com/scalameta/scalameta/releases/tag/v4.0.0-M1 ?